core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -734,9 +740,9 @@ private[spark] class BlockManager(
    * @param keepReadLock if true, this method will hold the read lock when it returns (even if the
    *                     block already exists). If false, this method will hold no locks when it
    *                     returns.
-   * @return `Some(PutResult)` if the block did not exist and could not be successfully cached,
-   *         or None if the block already existed or was successfully stored (fully consuming
-   *         the input data / input iterator).
+   * @return `Some(Option[Iterator[Any])` if the block did not exist and could not be successfully
+   *          cached, or None if the block already existed or was successfully stored (fully
+   *          consuming the input data / input iterator).
```

Comment:
this paragraph is a little too low level. I would start by explaining what the return value means on a high level, e.g.
```
@return iterator of values if the put failed. This is defined only if ...
```
then go into what `Some` or `None` means.